pkgs/development/tools/zigmod/buildZigmodPackage.nix

```diff
@@ -0,0 +1,76 @@
+{ lib
+, stdenv
+, cacert
+, git
+, zig
+, zigmod
+}:
+
+{ buildPhase ? ''
+    runHook preBuild
+
+    zigmod fetch
+    zig build -Drelease-safe -Dcpu=baseline --prefix $out
+  ''
+, fetchOutputHash
+, meta ? { }
+, nativeBuildInputs ? [ ]
+, pname
+, postUnpack ? ""
+, src
+, version
```

Comment:
This is not very extensible. Please take a look from https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/interpreters/python/mk-python-derivation.nix

Reply:
Switched to the @-pattern for the output derivation.